File: spring-boot-java/src/main/java/io/ssaragi/java/spring/boot/CustomExceptionHandler.java
package io.ssaragi.java.spring.boot;

import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.servlet.NoHandlerFoundException;
import org.springframework.http.ResponseEntity;

@ControllerAdvice
public class CustomExceptionHandler {

    @ExceptionHandler(NoHandlerFoundException.class)
    public ResponseEntity<String> handleNoHandlerFoundException(NoHandlerFoundException e) {
        return ResponseEntity.notFound().build();
    }
}

File: spring-boot-kotlin/src/main/kotlin/io/ssaragi/kotlin/spring/boot/CustomExceptionHandler.kt
package io.ssaragi.kotlin.spring.boot

import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.servlet.NoHandlerFoundException
import org.springframework.http.ResponseEntity

@ControllerAdvice
class CustomExceptionHandler {

    @ExceptionHandler(NoHandlerFoundException::class)
    fun handleNoHandlerFoundException(e: NoHandlerFoundException): ResponseEntity<String> {
        return ResponseEntity.notFound().build()
    }
}
